sdk/python/kfp/compiler/compiler.py

```diff
@@ -946,4 +940,32 @@ def _create_and_write_workflow(
         params_list,
         pipeline_conf)
     self._write_workflow(workflow, package_path)
+    _validate_workflow(workflow)
+
+
+def _validate_workflow(workflow: dict):
+  workflow = workflow.copy()
+  # Working around Argo lint issue
+  for argument in workflow['spec'].get('arguments', {}).get('parameters', []):
+    if 'value' not in argument:
+      argument['value'] = ''
 
+  yaml_text = yaml.dump(workflow)
+  if '{{pipelineparam' in yaml_text:
+    raise RuntimeError(
+        '''Internal compiler error: Found unresolved PipelineParam.
+Please create a new issue at https://github.com/kubeflow/pipelines/issues attaching the pipeline code and the pipeline package.'''
+    )
+
+  # Running Argo lint if available
+  import shutil
+  import subprocess
+  argo_path = shutil.which('argo')
+  if argo_path:
+    result = subprocess.run([argo_path, 'lint', '/dev/stdin'], input=yaml_text, encoding='utf-8', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Did it (maybe intentionally or accidentally) drop Python 3.5 support, because subprocess.run() does not seem to have `encoding` arg in python 3.5 [1]. As a result, it would cause TypeError.

[1]  https://docs.python.org/3.5/library/subprocess.html#subprocess.run

Reply:
Interesting... In Travis CI [1] we do test against Python 3.5 and it does not complain about this change. Will take a closer look to it.

Did you find anything wrong @ucdmkt ?


[1] https://github.com/kubeflow/pipelines/blob/master/.travis.yml#L54

Reply:
Okay in Travis, the unit tests we run `_compile()` instead of `compile()` Thus this line was not covered.

Reply:
> Okay in Travis, the unit tests we run `_compile()` instead of `compile()` Thus this line was not covered.

Took back my word. It's because in Travis Argo CLI is missing so it's not triggered.

Reply:
@ucdmkt Thanks for reporting this! We've fixed the bug in https://github.com/kubeflow/pipelines/pull/3122